vars/getBuildManifestInfo.groovy

```diff
@@ -0,0 +1,11 @@
+String[] call(Map args = [:]) {
+    def lib = library(identifier: "jenkins@20211123", retriever: legacySCM(scm))
+    String buildManifestUrl = args.buildManifestUrl ?: BUILD_MANIFEST_URL
+
+    String buildManifestPath = 'manifest.yml'
+    sh "wget ${buildManifestUrl} -O ${buildManifestPath}"
+    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: buildManifestPath))
+    String buildId = buildManifest.getArtifactBuildId()
+
+    return [buildManifestPath, buildId]
+}
```

Comment:
Returning a tuple limits the utility of this shared function. As suggested above, maybe something like `downloadBuildManifest` that downloads the remote manifest and returns the `buildManifest` object, then all the fields are available to the caller.

Reply:
Agreed this is a better approach, changed to `downloadBuildManifest()` in https://github.com/opensearch-project/opensearch-build/pull/1429/commits/c60435a14f764a93afa35f0ab1b002b3e93750a5